src/duckling/canvas/map-editor.component.ts

```diff
@@ -0,0 +1,67 @@
+import {
+    Component
+} from 'angular2/core';
+import {DisplayObject} from 'pixi.js';
+import {Observable} from 'rxjs';
+
+import {ArraySelect, SelectOption} from '../controls';
+import {EntitySystemService} from '../entitysystem/';
+import {Canvas} from './canvas.component';
+import {EntityDrawerService} from './drawing/entity-drawer.service';
+import {EntityCreatorTool, BaseTool, EntityMoveTool, TOOL_PROVIDERS} from './tools';
+
+/**
+ * The MapEditorComponent contains the canvas and tools needed to interact with the map.
+ */
+@Component({
+    selector: "dk-map-editor",
+    directives: [Canvas, ArraySelect],
+    viewProviders : [TOOL_PROVIDERS],
+    template: `
+        <dk-array-select
+            [value]="selectedTool"
+            [options]="toolOptions"
+            (selection)="onToolSelected($event)">
+        </dk-array-select>
+
+        <dk-canvas
+            [tool]="tool"
+            [width]="width"
+            [height]="height"
+            [stage]="mapStage">
+        </dk-canvas>
+    `
+})
+export class MapEditorComponent {
+    width : number = 500
+    height : number = 400;
+    mapStage : DisplayObject;
+
+    tool : BaseTool;
+    selectedTool : string = "move";
+    toolOptions : SelectOption [] = [
```

Comment:
These should be implemented using the key and title getters in the specific tool implementation.